httpx/decoders.py

```diff
@@ -113,6 +115,8 @@ def flush(self) -> bytes:
                 self.decompressor.finish()
             return b""
         except brotli.error as exc:  # pragma: nocover
+            if str(exc) == "Decompression error: incomplete compressed stream.":
```

Comment:
Then, rather than trying to guess if we've seen a null-data error we can do...

```python
def flush(self) -> bytes:
    if not self.seen_data:
        return b""
    ...
```